pkg/controller/v1beta1/inferenceservice/reconcilers/deployment/deployment_reconciler.go

```diff
@@ -388,9 +388,9 @@ func (r *DeploymentReconciler) Reconcile(ctx context.Context) ([]*appsv1.Deploym
 
 			// To avoid the conflict between HPA and Deployment,
 			// we need to remove the Replicas field from the deployment spec
-			// For external autoscaler, it should not remove replicas
+			// For external or none autoscaler, it should not remove replicas
 			modDeployment := deployment.DeepCopy()
-			if modDeployment.Annotations[constants.AutoscalerClass] != string(constants.AutoscalerClassExternal) {
+			if modDeployment.Annotations[constants.AutoscalerClass] != string(constants.AutoscalerClassExternal) && modDeployment.Annotations[constants.AutoscalerClass] != string(constants.AutoscalerClassNone) {
```

Comment:
this is the same

Reply:
Same for my question in: https://github.com/kserve/kserve/pull/4363#discussion_r2055621196